### ISSUE TITLE
Remove tool_dependencies.xml for everything in Conda

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/tools-iuc#947]
We should push new install-able revisions of everything that has passing Conda tests that removes the tool_dependencies.xml files. This will avoid confusion that may occur if dependencies exist in both places and pushes the Conda revolution forward.

**Comment by martenson** [comment]
this imo needs Conda-enabled Galaxies shipped by default i.e. after 16.10?

**Comment by jmchilton** [comment]
Galaxy never shipped with tool shed enabled by default, I certainly don't think it should be a requirement now for this.

**Comment by natefoo** [comment]
Second. Although this would break TS installs for pre-conda Galaxies.

Viva la revolución!

**Comment by mvdbeek** [comment]
yes, please.

**Comment by martenson** [comment]
Throwing out many polished TS packages just because there is Conda resolver available feels like an emotional response and waste of people's work. On one-by-one basis - when we decide that Conda is superior requirement resolver - the switch and version bump make sense.

I also believe the confusion that is mentioned here should be primarily fixed on the admin's side as described in https://github.com/galaxyproject/galaxy/issues/2927 https://github.com/galaxyproject/galaxy/issues/2925 https://github.com/galaxyproject/galaxy/issues/2911 etc before we start implementing any measures of this radicality.

**Comment by martenson** [comment]
Also please see https://github.com/galaxyproject/galaxy/issues/2934 for just one of the problems that can easily happen.

**Comment by mvdbeek** [comment]
> Throwing out many polished TS packages just because there is Conda resolver available feels like an emotional response and waste of people's work.

There is something true about this, but don't forget the long compile times and the difficulties with nested repository dependencies -- just one problem in the chain and you'll be loosing hours of your days in the most useless fashion.

**Comment by martenson** [comment]
@mvdbeek Let's identify those packages (& all repos that depend on them) and make them Conda-only then. I applaud to that.

**Comment by mvdbeek** [comment]
> @mvdbeek Let's identify those packages (& all repos that depend on them) and make them Conda-only then. I applaud to that.

Right now things are OK, I'll obviously come back once things start to fail again. But of the top of my head I would say sratools tools and sailfish.

**Comment by martenson** [comment]
sra-tools have another problem. Since [this change](https://github.com/galaxyproject/tools-iuc/commit/8550e0c7389533bebc75d7a93b89a3e59fe77bcd#diff-b132563ee6e9a7c534e9375c7d3f7e22R7) it includes all `<requirements>` for TS packages and for Conda (which are different). 

This means that when you install it using TS without Conda it will still show as missing dependencies despite working fine (because there is no sra-tools package in TS).

**Comment by mvdbeek** [comment]
Yup, there is currently no other way to handle diverging names for packages.
But from https://github.com/galaxyproject/galaxy/issues/2533 I gather that there is also a problem with the tool_dependencies.xml, that's why I'm thinking this would be a good candidate to have its tool_dependencies.xml removed.

**Comment by yhoogstrate** [comment]
I would prefer not rush with this. Yes, `tool_dependencies.xml` has issues but so has conda (e.g. https://github.com/conda/conda-build/pull/877).

I would be happiest to have some kind of transition/bug-report period before permanently migrating to conda. I would not be surprised if there are still galaxy instances around that are not having conda up and running (as mentioned by @natefoo).

**Comment by mvdbeek** [comment]
> I would prefer not rush with this. Yes, tool_dependencies.xml has issues but so has conda (e.g. conda/conda-build#877).

This issue is going away soon, and once this is the case we should choose a relatively long path in planemo to test tools, so that we are sure that all underlying dependencies have been rebuilt.

> I would be happiest to have some kind of transition/bug-report period before permanently migrating to conda. I would not be surprised if there are still galaxy instances around that are not having conda up and running (as mentioned by @natefoo).

Existing revisions will not be changed, and in other areas we are also advancing new features, like metadata column headers, tool profiles and so on. It has to be done at some point, so why not now? A significant fraction of instances that haven't updated until now will probably not have been updated in the next 6 months either, and I would guess that these instances will also be slow with updating their installed tools. But I agree, as long as the current tool_dependencies.xml has no issues, we should probably at least wait until 16.10 before we update tools just to get rid of the toolshed dependencies.
